### PR TITLE
Ensure Content-Length is set for Responses that need it

### PR DESCRIPTION
**Pull request by mfelsche**
For all ways to construct responses, ensure that they have a `Content-Length` header, adding one with `0` as value if none was provided.
Do not add one for responses with `Transfer-Encoding: chunked`.

This adds some overhead to building responses, but ensures (in a reasonable way) for legit API uses, that valid responses fall out in the end. Performance impact hasn't been measured yet, but can be done if desired.

This also included various internal fixes and some streamlining of response APIs. Overall I would consider this a non-backwards-compatible release.

Obsoletes #71

### CODE REVIEW

**Comment by ponylang-main** [comment]
Hi @mfelsche,

The changelog - fixed label was added to this pull request; all PRs with a changelog label need to have release notes included as part of the PR. If you haven't added release notes already, please do.

Release notes are added by creating a uniquely named file in the `.release-notes` directory. We suggest you call the file `74.md` to match the number of this pull request.

The basic format of the release notes (using markdown) should be:

```
## Title

End user description of changes, why it's important,
problems it solves etc.

If a breaking change, make sure to include 1 or more
examples what code would look like prior to this change
and how to update it to work after this change.
```

Thanks.

**Comment by SeanTAllen** [comment]
@mfelsche I think this should get 2 changelog entries. And two release notes. As this is pretty much two changes. 

1. the fix for including content-length for one shot.
2. the stuff that is "a breaking change"

can you manually add the slightly different and appropriate CHANGELOG entries. 1 to changed and 1 to fixed?

and manually add to next-release.md. different entries with names that match those in the changelog. the fixed should be simple, for the changed, detail what has changed about the API and what needs to be updated to adjust?

also you need to rebase it appears.

**Review by SeanTAllen — CHANGES_REQUESTED** [review]
I left a comment with "changes". Basically, the changelog entries and release notes entries.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,24 @@ All notable changes to this project will be documented in this file. This projec
 
 ### Fixed
 
+- Responses without `Transfer-Encoding: chunked` had no `Content-Length` header set when not done explicitly. This made some clients (e.g. curl) hang.
```
Comment (SeanTAllen) [inline CHANGELOG.md:9]: This isn't a changelog entry. It's very long. This should be a short title and link to the PR

Same for all the others.

File: .release-notes/next-release.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+## Response Creation API changes and Additions
+
+- `Response.transfer_coding()` changed to `Response.transfer_encoding()`.
+
+- `ResponseBuilderHeaders.set_content_length(content_length: USize)` has been added to set a content-length from a numeric value.
+- `ResponseBuilderBody.add_chunk()` now takes a `ByteSeq` instead of `Array[U8] val`. This allows for passing `String val` as well.
+
+- `BuildableResponse.create()` now only takes a `Status` and a `Version`. Content-Length and Transfer-Encoding can be set later with `set_content_length()` and `set_transfer_encoding()`
+- `BuildableResponse.delete_header(header_name: String)` was added to enable deletion of headers that have been set previously.
+- `BuildableResponse.set_transfer_coding()` changed to `BuildableResponse.set_transfer_encoding()`
```
Comment (SeanTAllen) [inline .release-notes/next-release.md:15]: Can you make these individual entries as appropriate with "before and after" code examples?

The name for each `##` entry should match up with an entry in the CHANGELOG.

File: CHANGELOG.md (hunk 1 above)
Comment (SeanTAllen) [reply CHANGELOG.md:9]: our standard is along the lines of this one I pulled as an example from the file:

```
- Update to ponylang/net_ssl 1.3.2 ([PR #69](https://github.com/ponylang/http_server/pull/69))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,12 +6,24 @@ All notable changes to this project will be documented in this file. This projec
...
+  - `ResponseBuilderHeaders.set_content_length(content_length: USize)` has been added to set a content-length from a numeric value.
+  - `BuildableResponse.delete_header(header_name: String)` was added to enable deletion of headers that have been set previously.
 
 ### Changed
 
+- The API for creating HTTP Responses changed slightly:
+  - `ResponseBuilderBody.add_chunk()` now takes a `ByteSeq` instead of `Array[U8] val`. This allows for passing `String val` as well.
+  - `BuildableResponse.create()` now only takes a `Status` and a `Version`. Content-Length and Transfer-Encoding can be set later with `set_content_length()` and `set_transfer_encoding()`
+  - `BuildableResponse.set_transfer_coding()` changed to `.set_transfer_encoding()`
+  - `Response.transfer_coding()` changed to `.transfer_encoding()`.
```
Comment (SeanTAllen) [inline CHANGELOG.md:25]: I'm not sure why the changelog tool is accepting these as valid. I'll have to look into it.

**Pull request merged**